lib/backup/s3remote/s3.go

```diff
@@ -302,10 +305,10 @@ func (fs *FS) HasFile(filePath string) (bool, error) {
 		Bucket: aws.String(fs.Bucket),
 		Key:    aws.String(path),
 	}
-	o, err := fs.s3.GetObject(input)
+	o, err := fs.s3.GetObject(context.Background(), input)
 	if err != nil {
-		var ae awserr.Error
-		if errors.As(err, &ae) && ae.Code() == s3.ErrCodeNoSuchKey {
+		var nsk *types.NoSuchKey
+		if errors.As(err, &nsk) {
```

Comment:
Probably, it would be better using `errors.Is()` here, since `nsk` isn't used here.